lib/types/application/ApplicationFormatter.js

```diff
@@ -19,12 +26,33 @@ class ApplicationFormatter extends AbstractFormatter {
 				"/": project.resources.configuration.paths.webapp
 			};
 
-			return this.readManifest(project).then(function(manifest) {
-				if (!manifest["sap.app"] || !manifest["sap.app"].id) {
+			return this.readManifest(project).then((manifest) => {
+				// check for a proper sap.app/id in manifest.json to determine namespace
+				const appId = manifest["sap.app"] && manifest["sap.app"].id;
+				if (!appId) {
 					log.warn(`No "sap.app" ID configuration found in manifest of project ${project.metadata.name}`);
 					return;
 				}
-				project.metadata.namespace = manifest["sap.app"].id.replace(/\./g, "/");
+				// check app id for placeholder and try to read the placeholder from the pom.xml
+				const parts = appId.match(/^\$\{(.*)\}$/);
```

Comment:
What about the idea of checking **${project.artifactId}** explicitly and maybe throw error in case of any other placeholder?

Reply:
No, there could be any other placeholder be used for that!